src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -350,6 +368,8 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
             }
 
             var parameters = DelegateParameters(delegateType);
+            CheckDelegateDynamicParametercs(parameters, diagnostics);
```

Comment:
Agree.  Overload changes would be an unintended consequence of this change.  I will adjust the code to not error here and check for the existance of the ctors in the emit code.